packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -124,8 +133,9 @@ export class InitError extends GOVUKFrontendError {
  *
  * @internal
  * @typedef {object} ElementErrorOptions
- * @property {string} componentName - The name of the component throwing the error
+ * @property {string} [componentName] - The name of the component throwing the error
  * @property {string} identifier - An identifier that'll let the user understand which element has an error. This is whatever makes the most sense
  * @property {Element | null} [element] - The element in error
  * @property {string} [expectedType] - The type that was expected for the identifier
+ * @property {import("../init.mjs").CompatibleClass} [component] - Component throwing the error
```

Comment:
**suggestion** Similarly to the `formatErrorMessage` helper, I think we only really need `{moduleName: string}` as a type here to avoid requesting more than what we'll use.

Maybe we need some kind of `HasModuleName` type as `{moduleName: string}` is a type we use both:
- in the `CompatibleClass`
- to cast the `ChildClassConstructor`
- to format the error message

How does that sound to you?